serve.py: Define logger if undefined

arenevier opened the pull request:
create_daemon is sometimes called in a new python instance, in which
case it looses the reference to global variable logger. When it happens,
we redefine logger with at default value.

Fixes https://github.com/web-platform-tests/wpt/issues/28245

jgraham commented on tools/serve/serve.py:
```diff
@@ -514,6 +514,12 @@ def start(self, init_func, host, port, paths, routes, bind_address, config, **kw
     def create_daemon(self, init_func, host, port, paths, routes, bind_address,
                       config, **kwargs):
+
+        global logger
```
I think you need to call `set_logger(config.logger)`. Maybe we could just have an `ensure_logger(config)` function.

arenevier replied:
done

jgraham replied:
This looks like you forgot to update the patch?

arenevier replied:
Indeed. I have pushed the correct patch now.

Commit:
serve.py: Define logger if undefined

create_daemon is sometimes called in a new python instance, in which
case it looses the reference to global variable logger. When it happens,
we redefine logger with at default value.

Fixes https://github.com/web-platform-tests/wpt/issues/28245
